internal/cmd/grpc.go

```diff
@@ -262,6 +258,56 @@ func NewGRPCServer(
 		logger.Debug("cache enabled", zap.Stringer("backend", cacher))
 	}
 
+	// Audit sinks configuration.
+	sinks := make([]auditsink.AuditSink, 0)
+
+	if cfg.Audit.Sinks.LogFile.Enabled {
+		logFileSink, err := logfile.NewLogFileSink(logger, cfg.Audit.Sinks.LogFile.Path)
+		if err != nil {
+			return nil, fmt.Errorf("opening file at path: %s", cfg.Audit.Sinks.LogFile.Path)
+		}
+
+		sinks = append(sinks, logFileSink)
+	}
+
+	// Based on audit sink configuration from the user, provision the audit sinks and add them to a slice,
+	// and if the slice has a non-zero length, add the audit sink interceptor.
+	if len(sinks) > 0 {
+		sse := auditsink.NewSinkSpanExporter(logger, sinks)
+		_, ok := tracingProvider.(*tracesdk.TracerProvider)
+		if !ok {
+			tracingProvider = tracesdk.NewTracerProvider(
+				tracesdk.WithBatcher(
+					sse,
+					tracesdk.WithBatchTimeout(cfg.Audit.Buffer.FlushPeriod),
+					tracesdk.WithMaxExportBatchSize(cfg.Audit.Buffer.Capacity),
+				),
+				tracesdk.WithResource(resource.NewWithAttributes(
+					semconv.SchemaURL,
+					semconv.ServiceNameKey.String("flipt"),
+					semconv.ServiceVersionKey.String(info.Version),
+				)),
+				tracesdk.WithSampler(tracesdk.AlwaysSample()),
+			)
+		} else {
+			tracingProvider.RegisterSpanProcessor(tracesdk.NewBatchSpanProcessor(sse, tracesdk.WithBatchTimeout(cfg.Audit.Buffer.FlushPeriod), tracesdk.WithMaxExportBatchSize(cfg.Audit.Buffer.Capacity)))
```

Comment:
It might be possible to tidy this all up by first making the tracing provider a `tracesdk.NewTracerProvider()` (instead of a noop). But without registering any span processors on it.

The just doing what you have here in the `else` case, and updating the earlier transport setting too to do the same.
So in both cases you're calling `tracingProvider.RegisterSpaceProcessor`.
In order to avoid casting and duplication.

Reply:
@GeorgeMac Done. I realized there aren't any crazy extra resources allocated with just using `tracingProvider` directly (unless you register a `SpanProcessor`). Made the change.